src/sas/system/config/config.py

```diff
@@ -0,0 +1,196 @@
+backwards compatible with the new config. Extreme caution should be
+exercised - when changing the config, don't just think about the
+problem at hand, but about the future maintainability of SasView in
+general.
+
+Adding to the Config class:
+Before adding a variable, think about whether it might more properly
+belong somewhere else, perhaps in the web or legal classes in the
+system package.
+Remember that config variables are accessed across the whole of sasview
+and that names need to be suitably descriptive.
+
+Deleting from the Config class:
+Currently (02/09/2022) the consequences of providing an entry in a
+config file that is not properly reflected in the Config class is a
+error. To ease backward compatibility, it is possible to disable
+the errors for a deleted class member by adding their name to
+the `_deleted_attributes` list. The suggested deletion process would
+be
+
+```
+[-]   self.my_variable = 10
+[+]   self._deleted_attributes.append("my_variable")
+```
+
+At major releases, additions to _deleted_attributes should be removed.
+
+
+Other Design Decisions made for the Config Class
+================================================
+
+The Config class cannot be dynamically modified, this prevents the config
+from having fields that are unspecified in the base class,
+and makes it so that all usages of fields can be automatically tracked.
+
+Subclassing of Config is also disabled for similar reasons.
+
+I have opted not to use frozen dataclasses at this time
+because, as they currently work, the behaviour would make creating
+different configs difficult.
+"""
+
+from sas.system.config.config_meta import ConfigBase, ConfigMeta
+
+
+class Config(ConfigBase, metaclass=ConfigMeta):
+
+    def __init__(self):
+        super().__init__()
+
+        # Flag for automated testing
+        # self.__TEST__ = False
+
+        # edit the list of file state your plugin can read
+        self.APPLICATION_WLIST = 'SasView files (*.svs)|*.svs'
+        self.APPLICATION_STATE_EXTENSION = '.svs'
+        self.PLUGIN_STATE_EXTENSIONS = ['.fitv', '.inv', '.prv', '.crf']
+        self.PLUGINS_WLIST = [
+            'Fitting files (*.fitv)|*.fitv',
+            'Invariant files (*.inv)|*.inv',
+            'P(r) files (*.prv)|*.prv',
+            'Corfunc files (*.crf)|*.crf']
+
+        self.ANALYSIS_TYPES = [
+            'Fitting files (*.fitv)',
+            'Invariant files (*.inv)',
+            'P(r) files (*.prv)',
+            'Corfunc files (*.crf)']
+
+        self.SHOW_WELCOME_PANEL = False
+        # OPEN and SAVE project menu
+        self.OPEN_SAVE_PROJECT_MENU = True
+        # VIEW MENU
+        self.VIEW_MENU = True
+        # EDIT MENU
+        self.EDIT_MENU = True
+
+
+
+        # OpenCL option - should be a string, either, "none", a number, or pair of form "A:B"
+        self.SAS_OPENCL = "none"
+
+        self.DEFAULT_OPEN_FOLDER = ""
+        self.TOOLBAR_SHOW = True
```

Comment:
On second glance, many of these booleans aren't saved in config.json and probably should. Here is an initial list that I will update as I find more and as they are fixed.

- TOOLBAR_SHOW
- VIEW_MENU
- whatever is used for showing/hiding the console log
- SHOW_WELCOME_PANEL is set to False in the config, but the welcome panel is still shown

Reply:
It won't get saved in config.json unless its been saved as something other than the default value.

Reply:
The welcome panel isn't what you think it is

Reply:
I hid the toolbar, console log, and data explorer, all of which are shown by default. I closed the program, but none of those changes were stored in config.json, so they were all visible when I opened the program again.

Reply:
> The welcome panel isn't what you think it is

You're right. I mixed up the splash screen and welcome panel.

Reply:
I will make a new issue for these

Reply:
#2208